src/effects/VST/VSTEffect.h

```diff
@@ -199,6 +210,65 @@ struct VSTEffectWrapper : public VSTEffectLink, public XMLTagHandler
    void SaveFXP(const wxFileName& fn) const;
    void SaveFXProgram(wxMemoryBuffer& buf, int index) const;
 
+   // VST plugin -> host callback
+   static intptr_t AudioMaster(AEffect *effect,
+                               int32_t opcode,
+                               int32_t index,
+                               intptr_t value,
+                               void * ptr,
+                               float opt);
+
+   // These are needed to move the AudioMaster callback from VSTEffect to this struct
+   //
+   intptr_t mCurrentEffectID {};
+   virtual void NeedIdle() = 0;
+   virtual void UpdateDisplay() = 0;
+   virtual VstTimeInfo* GetTimeInfo() = 0;
+   virtual void SetBufferDelay(int samples) = 0;
+   virtual float GetSampleRate() = 0;
+   virtual int GetProcessLevel() = 0;
+   virtual void SizeWindow(int w, int h) = 0;
+   virtual void Automate(int index, float value) = 0;
+
+   bool Load();
+   PluginPath   mPath;
+
+   // Define a manager class for a handle to a module
+#if defined(__WXMSW__)
+   using ModuleHandle = std::unique_ptr<wxDynamicLibrary>;
+#else
+   struct ModuleDeleter {
+      void operator() (void*) const;
+   };
+   using ModuleHandle = std::unique_ptr < char, ModuleDeleter >;
+#endif
+
+   ModuleHandle mModule{};
+
+   wxString mVendor;
+   wxString mDescription;
+   int      mVersion;
+   bool     mInteractive{ false };
+   unsigned mAudioIns{ 0 };
+   unsigned mAudioOuts{ 0 };
+   int      mMidiIns{ 0 };
+   int      mMidiOuts{ 0 };
+   bool     mAutomatable;
+
+   virtual void callSetProgram(int index) = 0;
+   virtual void Unload() = 0;
+
+
+   bool mUseLatency{ true };
+   int  mBufferDelay{ 0 };
+
+   sampleCount GetLatencyB() const
+   {
+      if (mUseLatency)
+         return mBufferDelay;
+      return 0;
```

Comment:
Last comment for this round of review!

This definition of `GetLatencyB` seems weird to me.  I think you do it because the `SetBufferDelay` virtual function must be defined for all Wrappers.  But I say again, make it non-pure and let it do nothing in the base class.  Let the Instance override it properly.  Then, that leaves no need to move the two data members into the wrapper.  They belong in the Instance only.

Reply:
I commited part of your suggestion in 4.5. `mUseLatency` can not be moved to the instance, because it is referenced by `VSTEffect::ShowOptions` - any idea on how to solve that?

Reply:
Just duplicate that boolean in the instance.  Make it a constructor argument.

Reply:
In commit 4.6 I moved `mUseLatency` and `mBufferDelay` in the instance, leaving a copy of `mUseLatency` in the Effect class which is passed to the instance at construction time.